include/hdf4cpp/HdfObject.h

```diff
@@ -38,14 +42,24 @@ class HdfObject {
     /// then the destructor of the destroyer will call the end access function.
     class HdfDestroyerChain {
       public:
-        HdfDestroyerChain() {
-        }
-        HdfDestroyerChain(const HdfDestroyerChain &other)
+        HdfDestroyerChain() = default;
+        HdfDestroyerChain(const HdfDestroyerChain &other) noexcept
             : chain(other.chain) {
         }
+        HdfDestroyerChain(HdfDestroyerChain &&other) noexcept
+            : chain(std::move(other.chain)) {
+        }
+        HdfDestroyerChain &operator=(const HdfDestroyerChain &other) noexcept {
+            chain = other.chain; // copy
+            return *this;
+        }
+        HdfDestroyerChain &operator=(HdfDestroyerChain &&other) noexcept {
+            chain = std::move(other.chain);
+            return *this;
+        }
 
-        void pushBack(HdfDestroyer *destroyer) {
-            chain.push_back(std::shared_ptr<HdfDestroyer>(destroyer));
+        void emplaceBack(const std::function<int32(int32)> &endFunction, int32 id) {
+            chain.emplace_back(new HdfDestroyer(endFunction, id));
```

Comment:
possible alternative:
```c++
chain.emplace_back(std::make_unique<HdfDestroyer>(endFunction, id))
```

Reply:
https://github.com/AnthonyCalandra/modern-cpp-features#stdmake_unique

Reply:
`std::make_unique` is available since c++14

However, we are compiling with c++11:
```cmake
set(CMAKE_CXX_STANDARD 11)
```